CRM/Member/WorkflowMessage/Membership/Membership.php

```diff
@@ -145,10 +165,23 @@ private function addExampleData(GenericWorkflowMessage $messageTemplate, array $
    * @throws \CRM_Core_Exception
    */
   private function getPriceSet(): ?array {
-    return (array) PriceSet::get(FALSE)
-      ->addWhere('extends', '=', CRM_Core_Component::getComponentID('CiviMember'))
-      ->addOrderBy('is_quick_config', 'DESC')
-      ->execute()->indexBy('id');
+    if (!$this->priceSets) {
+      $priceSets = (array) PriceSet::get(FALSE)
+        ->addWhere('extends:name', 'CONTAINS', 'CiviMember')
+        ->addOrderBy('is_quick_config', 'DESC')
+        ->execute()->indexBy('id');
+      $priceSetEntities = PriceSetEntity::get(FALSE)
+        ->addWhere('price_set_id', 'IN', array_keys($priceSets))
+        ->addWhere('entity_table', '=', 'civicrm_contribution_page')
+        ->addOrderBy('entity_id')
+        ->addSelect('price_set_id', 'entity_id')
+        ->execute();
+      foreach ($priceSetEntities as $priceSetEntity) {
+        $priceSets[$priceSetEntity['price_set_id']]['contribution_page_id'] = $priceSetEntity['entity_id'];
```

Comment:
just speaking hypothetically what happens if one price set is used in multiple contribution page id?

Reply:
Then it will get the last - which is fine - it just needs one

Reply:
ie - because this is just an example of the receipt then we are just looking for some 'life like' data